app/src/main/java/com/hedvig/app/feature/offer/OfferRepository.kt

```diff
@@ -8,38 +8,60 @@ import com.apollographql.apollo.coroutines.await
 import com.apollographql.apollo.coroutines.toFlow
 import com.apollographql.apollo.fetcher.ApolloResponseFetchers
 import com.hedvig.android.owldroid.fragment.CostFragment
+import com.hedvig.android.owldroid.fragment.QuoteBundleFragment
 import com.hedvig.android.owldroid.graphql.LastQuoteIdQuery
 import com.hedvig.android.owldroid.graphql.OfferQuery
+import com.hedvig.android.owldroid.graphql.QuoteCartSubscription
 import com.hedvig.android.owldroid.graphql.RedeemReferralCodeMutation
 import com.hedvig.android.owldroid.graphql.RemoveDiscountCodeMutation
+import com.hedvig.app.feature.offer.model.OfferModel
+import com.hedvig.app.feature.offer.model.toOfferModel
 import com.hedvig.app.util.LocaleManager
+import com.hedvig.app.util.apollo.QueryResult
+import com.hedvig.app.util.apollo.safeSubscription
+import com.hedvig.app.util.featureflags.Feature
+import com.hedvig.app.util.featureflags.FeatureManager
+import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.map
 
 class OfferRepository(
     private val apolloClient: ApolloClient,
     private val localeManager: LocaleManager,
+    private val featureManager: FeatureManager,
 ) {
     fun offerQuery(ids: List<String>) = OfferQuery(localeManager.defaultLocale(), ids)
 
-    fun offer(ids: List<String>) = apolloClient
-        .query(offerQuery(ids))
-        .watcher()
-        .toFlow()
-        .map(::toDataOrError)
+    fun offer(ids: List<String>): Flow<OfferResult> {
```

Comment:
This function now returns our own data model, instead of generated graphql classes. This means we can map both queries to the same model, and let a feature flag decide which query/subscription to use